lib/Test.rakumod

```diff
@@ -188,8 +188,14 @@ multi sub is(Mu $got, Mu:D $expected, $desc = '') is export {
                     ~ "     got: $got.raku()";
             }
             else {
+		if try { # if the type support Stringification
+		        # note: we can't use ^can('Str') as Buf error in its Str method itself
                 _diag "expected: '$expected'\n"
                     ~ "     got: '$got'";
+		} {} else {
```

Comment:
This empty block here is too confusing. Wouldn't it be better to pre-stringify `$expect` and `$got` and then use these values for `_diag`? Besides, `_diag` below already duplicates the one above.

Reply:
I was not sure if using a try {} CATCH {} structure made more sense. I don't want to refactor too much the code.

Reply:
It's not that much of a refactor. But taking care of code duplication is what must be done anyway.

Besides, instead of `try` you can test for `Buf` type object.

Reply:
I had another look at the code and I am not sure I want to pre build the string for diag (can be somehow costly for some object) and combining the 2 conditions will remove some clarity. I can't test for just Buf, you never know what class the user will have that can have the same behaviour. Duplicated code is not the best, but clarity is always better for me.
I could maybe change the first message to highlight the fact that it's only whitespaces that were differents in the 2 compared object.

Reply:
Good point about `Buf` not being the only possible cause of the problem. Still, as I looked closer at the code,  there is a suggestion and a fix to be made.

First, to the potential bug. It is not specced what `_diag` returns. Thus, it is possible that `try { }` construct would return a false value despite no exception was thrown. So, it'd be more correct with `try { _diag; True }`. 

The suggestion should improve the clarity by removing extra constructs: 

```perl6
try {
    _diag(...); 
    True 
} or _diag(...)
```

would do the same what `if ... else ...` currently does but has better overall readability.